File: ruby/testcode/weakcipher_007.rb
require_relative 'shared'
require 'openssl'

# vuln-code-snippet start ruby_weakcipher_blowfish
def encrypt_blowfish(req)
  plaintext = req.body_str
  cipher = OpenSSL::Cipher.new('BF-CBC') # vuln-code-snippet vuln-line ruby_weakcipher_blowfish
  cipher.encrypt
  cipher.key = cipher.random_key
  encrypted = cipher.update(plaintext) + cipher.final
  BenchmarkResponse.ok(encrypted.unpack1('H*'))
end
# vuln-code-snippet end ruby_weakcipher_blowfish

File: php/testcode/weakcipher_007.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakcipher_3des
function weakcipher007(BenchmarkRequest $req): BenchmarkResponse {
    $data = $req->post('data');
    $key = getenv('ENCRYPTION_KEY');
    $iv = random_bytes(8);
    $encrypted = openssl_encrypt($data, 'des-ede3-cbc', $key, 0, $iv); // vuln-code-snippet vuln-line php_weakcipher_3des
    return BenchmarkResponse::json(['ciphertext' => base64_encode($iv . $encrypted)]);
}
// vuln-code-snippet end php_weakcipher_3des
